src/lib/openjp2/jp2.c

```diff
@@ -2636,6 +2658,104 @@ static OPJ_BOOL opj_jp2_read_ftyp(opj_jp2_t *jp2,
     jp2->jp2_state |= JP2_STATE_FILE_TYPE;
 
     return OPJ_TRUE;
+}
+
+static OPJ_BOOL opj_jp2_read_asoc(	opj_jp2_t *jp2,
+									OPJ_BYTE * p_header_data,
+									OPJ_UINT32 p_header_size,
+									opj_event_mgr_t * p_manager)
+{
+    OPJ_UINT32 label_tag;
+    OPJ_UINT32 asoc_tag;
+    OPJ_UINT32 asoc_size;
+    opj_jp2_asoc_t *asoc;
+
+    /* preconditions */
+    assert(jp2 != 00);
+    assert(p_header_data != 00);
+    assert(p_manager != 00);
+
+    opj_read_bytes(p_header_data,&asoc_size,4);
+    p_header_data += 4;
+    p_header_size -= 4;
+
+    opj_read_bytes(p_header_data,&label_tag,4);
+    p_header_data += 4;
+    p_header_size -= 4;
+    asoc_size -= 4;

```

Comment:
you should validate asoc_size is >= 4

Reply:
and probably verify asoc_size against p_header_size to avoid excessive memory allocation attempt

Reply:
Fixed